chain/network/src/test_loop.rs

```diff
@@ -21,18 +14,33 @@ use crate::types::{
     NetworkRequests, NetworkResponses, PeerManagerMessageRequest, PeerManagerMessageResponse,
     SetChainInfo,
 };
+use near_async::actix::ActixResult;
+use near_async::futures::{FutureSpawner, FutureSpawnerExt};
+use near_async::messaging::{Actor, AsyncSender, CanSend, Handler, SendAsync, Sender};
+use near_async::time::Clock;
+use near_async::{MultiSend, MultiSenderFrom};
+use near_primitives::hash::CryptoHash;
+use near_primitives::network::PeerId;
+use near_primitives::types::AccountId;
 
 /// Subset of ClientSenderForNetwork required for the TestLoop network.
 /// We skip over the message handlers from view client.
 #[derive(Clone, MultiSend, MultiSenderFrom)]
 pub struct ClientSenderForTestLoopNetwork {
     pub block: AsyncSender<BlockResponse, ()>,
+    pub block_headers: AsyncSender<BlockHeadersResponse, ActixResult<BlockHeadersResponse>>,
     pub block_approval: AsyncSender<BlockApproval, ()>,
     pub transaction: AsyncSender<ProcessTxRequest, ProcessTxResponse>,
     pub chunk_endorsement: AsyncSender<ChunkEndorsementMessage, ()>,
 }
 
-type NetworkRequestHandler = Arc<dyn Fn(NetworkRequests) -> Option<NetworkRequests>>;
+#[derive(Clone, MultiSend, MultiSenderFrom)]
+pub struct ViewClientSenderForTestLoopNetwork {
+    pub block_headers_request: AsyncSender<BlockHeadersRequest, ActixResult<BlockHeadersRequest>>,
+    pub block_request: AsyncSender<BlockRequest, ActixResult<BlockRequest>>,
+}
+
+type NetworkRequestHandler = Box<dyn Fn(NetworkRequests) -> Option<NetworkRequests>>;
```

Comment:
Curious why did we change this from Arc to Box? Is there some rusty philosophical reason or both work?